javascript/ql/src/Security/CWE-020/IncompleteUrlSchemeCheck.ql

```diff
@@ -42,14 +91,14 @@ DataFlow::Node schemeCheck(DataFlow::Node nd, DangerousScheme scheme) {
 }
 
 /** Gets a data-flow node that checks an instance of `ap` against the given `scheme`. */
-DataFlow::Node schemeCheckOn(AccessPath ap, DangerousScheme scheme) {
-  result = schemeCheck(ap.getAnInstance().flow(), scheme)
+DataFlow::Node schemeCheckOn(DataFlow::SourceNode root, string path, DangerousScheme scheme) {
```

Comment:
Is this introduced in the name of performance?

Reply:
Actually no. One of the tests I added failed because the internal `AccessPath` class was too restrictive, so I switched. It's also better not to depend on an internal API.